essentials/src/collector/mod.rs

```diff
@@ -781,7 +795,9 @@ impl Collector {
 		// Fill and write dispute info structure
 		let dispute_info = DisputeInfo {
 			dispute: dispute_event.clone(),
-			initiated: self.state.current_relay_chain_block_number,
+			initiated: relay_block_number,
+			initiator_indices,
+			session_index: self.state.current_session_index,
```

Comment:
I would suggesting reading `session_index` from the `InherentData`, i.e. `extract_dispute_initiators` should return a tuple `(indices, session_index)`. Even if dispute is initiated in the current session, doesn't *necessarily* mean that validators indices are from the current session, they are from when the candidate was included, which may or may not be the current session.